tests/app/authentication/test_authenticator.py

```diff
@@ -17,7 +17,7 @@ def setUp(self):
         application.permanent_session_lifetime = timedelta(seconds=1)
         self.application = application
         # Use an in memory database
-        self.database = Database("sqlite://", 1, 0)
+        self.database = Database("sqlite", "", None, None, 1, 0)
```

Comment:
We should use named args when instantiating Database as it's unclear what all the arguments are for. It's also worth considering moving the retry count and retry delay values into constants in database.py